src/content/reference/react/cache.md

```diff
@@ -0,0 +1,485 @@
+
+#### Returns {/*returns*/}
+
+`cache` returns a cached version of `fn` with the same type signature. It does not call `fn` in the process.
+
+When calling `cachedFn` with given arguments, it first checks if a cached result exists in the cache. If a cached result exists, it returns the result. If not, it calls `fn` with the arguments, stores the result in the cache, and returns the result. The only time `fn` is called is when there is a cache miss.
+
+<Note>
+
+The optimization of caching return values based on inputs is known as [_memoization_](https://en.wikipedia.org/wiki/Memoization). We refer to the function returned from `cache` as a memoized function.
+
+</Note>
+
+#### Caveats {/*caveats*/}
+
+[//]: # 'TODO: add links to Server/Client Component reference once https://github.com/reactjs/react.dev/pull/6177 is merged'
+
+- React will invalidate the cache for all memoized functions for each server request. 
+- Each call to `cache` creates a new function. This means that calling `cache` with the same function multiple times will return different memoized functions that do not share the same cache.
+- `cachedFn` will also cache errors. If `fn` throws an error for certain arguments, it will be cached, and the same error is re-thrown when `cachedFn` is called with those same arguments.
+- `cache` is for use in [Server Components](https://react.dev/blog/2023/03/22/react-labs-what-we-have-been-working-on-march-2023#react-server-components) only. There are plans to introduce `cache` for Client Components, but it is not a supported feature today. There are plans to add a warning when `cache` is used in Client Components.
+
+---
+
+## Usage {/*usage*/}
+
+### Cache an expensive computation {/*cache-expensive-computation*/}
+
+Use `cache` to skip duplicate work.
+
+```js [[1, 7, "getUserMetrics(user)"],[2, 13, "getUserMetrics(user)"]]
+import {cache} from 'react';
+import calculateUserMetrics from 'lib/user';
+
+const getUserMetrics = cache(calculateUserMetrics);
+
+function Profile({user}) {
+  const metrics = getUserMetrics(user);
+  // ...
+}
+
+function TeamReport({users}) {
+  for (let user in users) {
+    const metrics = getUserMetrics(user);
+    // ...
+  }
+  // ...
+}
+```
+
+If the same `user` object is rendered in both `Profile` and `TeamReport`, the two components can share work and only call `calculateUserMetrics` once for that `user`. 
+
+Assume `Profile` is rendered first. It will call <CodeStep step={1}>`getUserMetrics`</CodeStep>, and check if there is a cached result. Since it is the first time `getUserMetrics` is called with that `user`, there will be a cache miss. `getUserMetrics` will then call `calculateUserMetrics` with that `user` and write the result to cache. 
+
+When `TeamReport` renders its list of `users` and reaches the same `user` object, it will call <CodeStep step={2}>`getUserMetrics`</CodeStep> and read the result from cache.
+
+<Pitfall>
```

Comment:
It feels like this is calling out two pitfalls at the same time:

1. calling cache inside a component will recreate the cached function (and the cache) on every render
2. you need to share the cached function across components or the cached won't be shared

I found it a bit difficult to follow. Have you considered factoring 1 out of this pitfall into a troubleshooting section? e.g. a title for a troubleshooting section based on 1 might be somethign like "My memoized function isn't caching my data fetch/computation"

Reply:
I will keep it for now. I had tried separating it out in a previous iteration, I think it won't be a major footgun as long as the larger pitfall is addressed. We can always update it if we see evidence of this being a common pattern